specs/TransportProtocols.md

```diff
@@ -0,0 +1,176 @@
+
+```
+foo: bar
+baz: boz
+baz: biz
+quz: qoz
+baz: flarg
+
+foo: boz
+
+```
+
+In the first event, the value of `baz` would be `boz\nbiz\nflarg`, due to the concatenation behavior above. Full details can be found in the spec linked above.
+
+In this transport, the client establishes an SSE connection to `[endpoint-base]/sse`, and the server responds with an HTTP response with a `Content-Type` of `text/event-stream`. Each SSE event represents a single frame from client to server. The transport uses unnamed events, which means only the `data` field is available. Thus we use the first line of the `data` field for frame metadata. The frame body starts on the **second** line of the `data` field value. The first line has the following format (Identifiers in square brackets `[]` indicate fields defined below):
+
+```
+[Type],[Fin]
+```
+
+* `[Type]` is a single UTF-8 character representing the type of the frame; `T` indicates Text, and `B` indicates Binary.
+* `[Fin]` is a single UTF-8 character indicating if the frame is the last frame of the message; `T` indicates `true`, `F` indicates `false`
+
+If the `[Type]` field is `T`, the remaining lines of the `data` field contain the value, in UTF-8 text. If the `[Type]` field is `B`, the remaining lines of the `data` field contain Base64-encoded binary data. Any `\n` characters in Binary frames are removed before Base64-decoding. However, servers should avoid line breaks in the Base64-encoded data.
+
+For example, when sending the following frames (`\n` indicates the actual Line Feed character, not an escape sequence):
+
+* Type=`Text`, Fin=`true`, "Hello\nWorld"
+* Type=`Binary`, Fin=`false`, `0x01 0x02`
+* Type=`Binary`, Fin=`true`, `0x03 0x04`
+
+The encoding will be as follows
+
+```
+data: T,T
+data: Hello
+data: World
+
+data: B,F
+data: AQI=
+
+data: B,T
+data: AwQ=
+
+```
+
+## Long Polling (Server-to-Client only)
+
+Long Polling is a server-to-client half-transport, so it is always paired with HTTP Post. It requires a connection already be established using the `[endpoint-base]/negotiate` endpoint.
+
+Long Polling requires that the client poll the server for new messages. Unlike traditional polling, if there are no messages available, the server will simply block the request waiting for messages to be dispatched. At some point, the server, client or an upstream proxy will likely terminate the connection, at which point the client should immediately re-send the request. Long Polling is the only transport that allows a "reconnection" where a new request can be received while the server believes an existing request is in process. This can happen because of a time out. When this happens, the existing request is immediately terminated with an empty result (even if messages are available) and the new request replaces it as the active poll request.
```

Comment:
> the server will simply block the request waiting for messages to be dispatched

Do we want to mention timeouts here? Long polling transports will have to timeout and re-poll even if there's no data to deal with proxies that kill requests.